torch/cuda/__init__.py

```diff
@@ -103,18 +103,30 @@ def _check_capability():
     PyTorch no longer supports this GPU because it is too old.
     """
 
+    weak_gpu_warn = """
+    Found GPU%d %s which has limited memory %d.
+    This device should probably be excluded by editing the CUDA_VISIBLE_DEVICES
+    environment variable.
+    """
+
     CUDA_VERSION = torch._C._cuda_getCompiledVersion()
     for d in range(device_count()):
+        name = get_device_name(d)
+
         capability = get_device_capability(d)
         major = capability[0]
-        name = get_device_name(d)
         if CUDA_VERSION < 8000 and major >= 6:
             warnings.warn(incorrect_binary_warn % (d, name, 8000, CUDA_VERSION))
         elif CUDA_VERSION < 9000 and major >= 7:
             warnings.warn(incorrect_binary_warn % (d, name, 9000, CUDA_VERSION))
         elif capability == (3, 0) or capability == (5, 0) or major < 3:
             warnings.warn(old_gpu_warn % (d, name, major, capability[1]))
 
+        total_mem = get_total_memory(d)
+        if total_mem < int(4e9):
```

Comment:
We should warn about imbalance, not about not having top-of-the-shelf GPUs

Reply:
see my response below